doc/release_notes.rst

```diff
@@ -13,6 +13,8 @@ This part of documentation collects descriptive release notes to capture the mai
 
 **New Features and Major Changes**
 
+* Add pseudo `branch()` to streamline snakemake workflow. Replace this with Snakemake implementation if Snakemake >= 8.3.0. Added an option to configure which sector components to include which when disabled, irrelevant rules are skipped automatically `PR #1583 <https://github.com/pypsa-meets-earth/pypsa-earth/pull/1538>`__
```

Comment:
A very minor thing: would be good to fix a typo in the PR number used for the cross-reference